extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TypeCheckExcludeBuildItem.java

```diff
@@ -16,15 +17,26 @@
 public final class TypeCheckExcludeBuildItem extends MultiBuildItem {
 
     private final Predicate<TypeCheck> predicate;
+    private final boolean extensionMethodPredicate;
 
     public TypeCheckExcludeBuildItem(Predicate<TypeCheck> predicate) {
         this.predicate = Objects.requireNonNull(predicate);
+        this.extensionMethodPredicate = false;
+    }
+
+    public TypeCheckExcludeBuildItem(Predicate<TypeCheck> predicate, boolean extensionMethodPredicate) {
+        this.predicate = predicate;
+        this.extensionMethodPredicate = extensionMethodPredicate;
     }
 
     public Predicate<TypeCheck> getPredicate() {
         return predicate;
     }
 
+    public boolean isExtensionMethodPredicate() {
```

Comment:
It might be useful to add a note about what is this actually used for... WDYT?

Reply:
Sure, I added a comment, just let me know if it's not properly described. Thank you